build-scripts/Dockerfile.debian-build

```diff
@@ -0,0 +1,99 @@
+# NOTE: we have to add `--privileged` to `docker run` to be able to debug inside the container, or else we get a pretty cryptic error once we `run` inside the debugger
+# NOTE: we can probably run an external debugger and attach it to the process inside the container somehow, to be able to debug inside the container using a full IDE on the host
+RUN apt-fast -qq install --no-install-recommends lldb-6.0
+
+
+# we prefer to build as a regular user (cdda) rather than root
+RUN mkdir /cdda
+# --disabled-password is a way to rush through the user generation (it's inelegant though), and then we set the password 'lol'
+#TODO: bring the password in as an ARG with an option
+RUN adduser cdda --disabled-password && echo cdda:lol | chpasswd
+USER cdda
+WORKDIR /home/cdda
+
+# copy everything from the source directory on the host (the `.` in docker build) - except the contents of .dockerignore
+COPY --chown=cdda:cdda . .
+
+# we build in build dirs rather than at the repo root
+# TODO: move build dirs to a directory ABOVE the source so we can Docker mount in the source dir for 'live' builds of code you're hacking on on the host
+RUN mkdir cmake-build-debug cmake-build-debug-tiles
+
+WORKDIR /home/cdda/cmake-build-debug
+
+
+WORKDIR /home/cdda/cmake-build-debug
+
+RUN env CC=clang-6.0 CXX=clang++-6.0 cmake ..
+
+RUN echo `pwd` configured for build; to actually build, run "make -j`nproc --all`" here
+
+WORKDIR /home/cdda/cmake-build-debug-tiles
+
+RUN env CC=clang-6.0 CXX=clang++-6.0 cmake -DTILES=ON ..
+
+RUN echo `pwd` configured for build; to actually build, run "make -j`nproc --all`" here
+
+
+## ssh server stuff (experimental and not needed except for testing ssh stuff with CDDA! you don't need ssh to get a shell inside the container!)
+## uncomment this to get ssh
+##
+## tell Docker to expose port 22 by default on `docker run` (the result is that it's mapped to a random port on the host so you can ssh in with ssh cdda@localhost:{RANDOMPORT}. Running `docker ps` will reveal the external port.
+## Note that you DO NOT need SSH to get a shell inside the container. This is for testing only.
+## Crash course: The usual way into the container is `docker {run or exec} -it {container/image name} bash` (or better yet, tmux or fish rather than bash). Docker run is to start a running container from an image, docker exec is to execute a process inside an already-started container.
+## TODO: expose 22 to fixed port by default, perhaps with an ARG?
+#EXPOSE 22
+##
+## the CMD entry tells Docker to run that command by default when you `docker run` the image as a container
+## NOTE: disabled as it's probably not a good default for the container. However, when enabled, `docker run` gives you a somewhat persistent container that accepts SSH connections, and then for a regular shell inside the container you do `docker exec -it {containername} bash` (or ssh if you wish)
```

Comment:
You can have a persistent container to `exec` into with
```
CMD ["tail", "-f", "/dev/null"]
```

Reply:
Hey that’s clever, thanks. Tailing the build / CDDA game log would be even better.